### DISCUSSION TITLE
Suggestion: regex support?

### DISCUSSION THREAD

**Discussion by spencer-michaels** [snezhig/obsidian-front-matter-title#61] [Ideas]
Hi! Amazing plugin, it's been super useful for my vault and my use of it! I was thinking, however, that something like regex support would be extremely useful. For instance, my note titles use a `YYMMDD.HHMM — concept or title` format, and it would be useful to have a regex filter it to just have the concept or title in tabs/file explorer! I'm able to do so in Dataview tables, the same here would be great. I could, of course, just write a small script to append the concept to my YAML headers, but having to tweak thousands of files is always risky...

**Comment by snezhig** [comment]
HI, thank you for the feedback, but I did not get the idea.
How is it supposed to work?

**Reply by spencer-michaels** [reply]
Kind of like what [this Reddit post](https://www.reddit.com/r/ObsidianMD/comments/y6hlls/any_plugins_to_apply_regex_filter_to_titles_as/) explains: using a Regex filter to show only a specific group. I put an example in [this regex101](https://regex101.com/r/i5NDdl/1) — having `YYMMDD.HHMM — title` display only the match, so as `title`.

**Reply by snezhig** [reply]
Do you mean using different templates for different managers?

**Reply by antomonte** [reply]
Hi!

I think that the idea of afk is that the plugin extracts the Title based on regex.

In his sample, given this filename/heading ...
YYMMDD.HHMM — this is my title
(actually YYMMDD.HHMM would be real dates, not just that fixed text)

... and adding to plugin regex support, for instance with this setting...
/((?:[^ —\r\n]*){2})$/gm

... it would came up with this result, thanks to the "black magic" of regex :) ...
this is my title

... which would actually be shown as Title at tab, file navigator, etc...

I think this is a really great idea, as your plugin offers a huge improvement on handling Titles.

**Reply by snezhig** [reply]
I agree that it is a great idea, so It may be added.

**Reply by snezhig** [reply]
It has been added at 2.8.0

**Reply by spencer-michaels** [reply]
Awesome! Exactly what I was hoping for — and I'm sure many other people will find it useful too! No compatibility issues with my (dozens) of plugins (other than an irrelevant minor issue with the Force Note View plugin that can be fixed by changing its settings, in case anyone else has the same problem). Thanks so much, you've solved one of my biggest issues with my vault!
